Ginger/Ginger/Actions/ActionConversion/SelectActionWzardPage.xaml.cs

```diff
@@ -101,10 +101,26 @@ private void SetGridView()
             xGridConvertibleActions.SetAllColumnsDefaultView(view);
             xGridConvertibleActions.InitViewItems();
             xGridConvertibleActions.SetTitleLightStyle = true;
+            xGridConvertibleActions.btnMarkAll.Visibility = System.Windows.Visibility.Visible;
+            xGridConvertibleActions.MarkUnMarkAllActive += MarkUnMarkAllActions;
             xGridConvertibleActions.ValidationRules = new List<ucGrid.eUcGridValidationRules>()
             {
                 ucGrid.eUcGridValidationRules.CheckedRowCount
             };
         }
+
+        private void MarkUnMarkAllActions(bool ActiveStatus)
```

Comment:
1- you do not need all of this IF checks
2- you do not need to pull the list from the Grid datasource you already know what is the list- it's mWizard.ActionToBeConverted
3- you do not need to re-enter the list again to the Grid.DataSource because you only changing some fields in it existing objects list, you don't changing the list

please replace all this code with below simple three rows
  foreach (ConvertableActionDetails act in mWizard.ActionToBeConverted)
                {
                    act.Selected = ActiveStatus;
                }

Reply:
Changed as suggested, but if condition will prevent the error if the list is null